alibuild_helpers/build_template.sh

```diff
@@ -104,13 +104,13 @@ cd "$BUILDDIR"
 #   the relocation script so that it takes into account the new location.
 if [[ "$CACHED_TARBALL" == "" && ! -f $BUILDROOT/log ]]; then
   set -o pipefail
-  bash -ex "$WORK_DIR/SPECS/$ARCHITECTURE/$PKGNAME/$PKGVERSION-$PKGREVISION/$PKGNAME.sh" 2>&1 | tee "$BUILDROOT/log"
+  . "$WORK_DIR/SPECS/$ARCHITECTURE/$PKGNAME/$PKGVERSION-$PKGREVISION/$PKGNAME.sh" 2>&1 | tee "$BUILDROOT/log"
```

Comment:
Hi @syamasak, we deliberately open a new Bash shell as we want a new environment. Sourcing the build script inside the current one is not exactly an option. We absolutely don't want a recipe to taint the rest of the environment. On top of that, where's `set -x` here?

However, I see an easy solution to that. You can try to amend this PR by wrapping script calls in `(...)`, so that you are effectively opening a new subshell with its separate environment.

Something like (and please use `source`, it's clearer than `.`):

```bash
 (set -x; source "$WORK_DIR/SPECS/$ARCHITECTURE/$PKGNAME/$PKGVERSION-$PKGREVISION/$PKGNAME.sh" 2>&1) | tee "$BUILDROOT/log"
```

The subshell should exit with an errorcode, and since we have already `set -o pipefail` and `set -e` this should cause the build template to exit with an error too.

Can you please implement my suggestion, and check if it effectively works by deliberately breaking a build recipe please?

Thanks!